web_chatt_filtering/static/src/js/chatter.js

```diff
@@ -0,0 +1,75 @@
+/* License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('mymail.Chatter', function (require) {
+    "use strict";
+
+    var Chatt = require('mail.Chatter');
+
+    Chatt.include({
+        events: _.extend({}, Chatt.prototype.events, {
+            'keyup input#search_partner': function (ev) {
+                if (ev.which === $.ui.keyCode.ENTER) {
+                    this.filter_message($(ev.target).val().trim());
+                }
+            },
+        }),
+
+        filter_message: function (substring) {
```

Comment:
Can you refactor this function so it's easier inheritable. E.g. put the rpc call inside a separate function.